spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationPropertySource.java

```diff
@@ -249,6 +272,56 @@ private Object createFeature(ConfigurationSetting item) throws IOException {
                 } else if (!featureItem.getEnabled()) {
                     return false;
                 }
+                mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
+                for (int filter = 0; filter < feature.getEnabledFor().size(); filter++) {
+                    if (feature.getEnabledFor().get(filter).getName().equals(TARGETING_FILTER)) {
+                        FeatureFilterEvaluationContext context = feature.getEnabledFor().get(filter);
+
+                        LinkedHashMap<String, Object> parameters = context.getParameters();
+
+                        if (parameters != null) {
+                            Object audienceObject = parameters.get(AUDIENCE);
+                            if (audienceObject != null) {
+                                parameters = (LinkedHashMap<String, Object>) audienceObject;
+                            }
+
+                            List<Object> users = new ArrayList<Object>();
+                            List<Object> groupRollouts = new ArrayList<Object>();
+
+                            users = mapper.convertValue(parameters.get(USERS_CAPS), users.getClass());
+                            groupRollouts = mapper.convertValue(parameters.get(GROUPS_CAPS), groupRollouts.getClass());
+
+                            Map<String, Object> userMap = new HashMap<String, Object>();
+                            Map<String, Object> rolloutMap = new HashMap<String, Object>();
+
+                            if (users != null) {
+                                for (int i = 0; i < users.size(); i++) {
+                                    userMap.put(String.valueOf(i), users.get(i));
+                                }
+                            }
+                            if (groupRollouts != null) {
+                                for (int i = 0; i < groupRollouts.size(); i++) {
+                                    rolloutMap.put(String.valueOf(i), groupRollouts.get(i));
+                                }
+                            }
+
+                            parameters.put(USERS, userMap);
+                            parameters.remove(USERS_CAPS);
+
+                            parameters.put(GROUPS, rolloutMap);
+                            parameters.remove(GROUPS_CAPS);
+
+                            parameters.put(DEFAULT_ROLLOUT_PERCENTAGE, parameters.get(DEFAULT_ROLLOUT_PERCENTAGE_CAPS));
+                            parameters.remove(DEFAULT_ROLLOUT_PERCENTAGE_CAPS);
+
+                            context.setParameters(parameters);
+                            HashMap<Integer, FeatureFilterEvaluationContext> enabledFor = feature.getEnabledFor();
+                            enabledFor.put(filter, context);
+                            feature.setEnabledFor(enabledFor);
+                        }
+                    }
+                }
+                mapper.configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, false);
```

Comment:
I did some code suggestions to try to simplify the logic and complexity. Let me know what do you think:
https://github.com/samukce/spring-cloud-azure/blob/15bf66d845e4c05d99020517671415f41f36cf9f/spring-cloud-azure-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationPropertySource.java#L243-L294